### PR TITLE
Feat/config repostable reaction

### PR DESCRIPTION
**Pull request by haongo138**
**What does this PR do?**

-   Add repost reaction configuration cmds & implement tracking repostable messages

### CODE REVIEW

File: pkg/handler/webhook.go [hunk 1]
```diff
@@ -206,3 +208,44 @@ func (h *Handler) handleGuildCreate(c *gin.Context, data json.RawMessage) {
...
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	conf, err := h.entities.GetGuildRepostReactionConfigByReaction(req.GuildID, req.Reaction)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	if req.ReactionCount < conf.Quantity {
```
Comment (anhnh12) [inline pkg/handler/webhook.go:231]: why do we return error here? we should ignore and return if it hasn't reached the required quantity to repost.
can put a log here to track if necessary

File: pkg/handler/webhook.go [hunk 2]
```diff
@@ -206,3 +208,44 @@ func (h *Handler) handleGuildCreate(c *gin.Context, data json.RawMessage) {
...
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "insufficient quantity"})
+		return
+	}
+
+	if repostable := h.entities.IsRepostableMessage(req.GuildID, req.MessageID); !repostable {
```
Comment (anhnh12) [inline pkg/handler/webhook.go:236]: same as above, ignore and return if users react to normal messages

Comment (anhnh12) [inline pkg/handler/webhook.go:236]: also why do we need this check?

Comment (haongo138) [reply pkg/handler/webhook.go:236]: for the question, a message should be reposted once, so i want to check from history whether a message was previously reposted or not

File: migrations/schemas/20220611111531-message_repost_histories.sql [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+-- +migrate Up
+create table message_repost_histories
```
Comment (anhnh12) [inline migrations/schemas/20220611111531-message_repost_histories.sql:3]: why do we need to store repost history

Comment (haongo138) [reply migrations/schemas/20220611111531-message_repost_histories.sql:3]: i want to track that a message was already reposted or not, so when users keep reacting/removing on that message, it'll not be reposted again

**Pull request merged**